include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -555,13 +555,9 @@ MatrixFree<dim, Number, VectorizedArrayType>::is_supported(
     return false;
 
   // then check of the base element is supported
-  if (dynamic_cast<const FE_Poly<TensorProductPolynomials<dim>, dim, spacedim>
-                     *>(fe_ptr) != nullptr)
+  if (dynamic_cast<const FE_Poly<dim, spacedim> *>(fe_ptr) != nullptr)
```

Comment:
@masterleinad how to I check to make sure this is an `FE_Poly` class with a supported `poly_space`? I thought I could try something like below, but it doesn't build because `poly_space` is protected.
```suggestion
    if (dynamic_cast<const FE_Poly<dim, spacedim> *>(fe_ptr) != nullptr)
  {
	const FE_Poly<dim, spacedim> *fe_poly_ptr = dynamic_cast<const FE_Poly<dim, spacedim> *>(fe_ptr);
	if(dynamic_cast<TensorProductPolynomials<dim> *>(fe_poly_ptr->poly_space.get()) != nullptr)
	  return true;
    if(dynamic_cast<TensorProductPolynomials<dim> *>(fe_poly_ptr->poly_space.get()) != nullptr)
	  return true;
  }
```